mobile/src/controllers/UserEventsController.ts

```diff
@@ -0,0 +1,106 @@
+import { observable, toJS, transaction } from 'mobx';
+import RepoFactory from 'common/controllers/RepoFactory';
+import { transferChangedFields } from 'common/utils/fields';
+import { ThrottleAction } from 'common/utils/throttle';
+import { IEvent, Event } from 'common/utils/event';
+import { Timestamped, UserEvents } from 'common/models/UserEvents';
+import { DomainSlug } from 'src/constants/Domain';
+
+export interface IUserEventsController {
+  readonly current: Readonly<UserEvents>;
+  readonly synced: IEvent;
+
+  logEvent(event: keyof UserEvents, newValue: Timestamped<string>): void;
+
+  flushChanges(): Promise<void>;
+}
+
+export class UserEventsController implements IUserEventsController {
+
+  @observable
+  private _current: UserEvents = null;
+
+  private _uid: string;
+  private readonly _syncThrottle = new ThrottleAction<Promise<void>>(200);
+
+  private readonly _synced = new Event();
+
+  public get current(): Readonly<UserEvents> { return this._current; }
+  public get synced(): IEvent { return this._synced; }
+
+  async load(uid: string) {
+    if (this._uid === uid) {
+      return;
+    }
+
+    this._uid = uid;
+    this._current = await RepoFactory.Instance.userEvents.getByUserId(uid);
+
+    let updateDiff: Partial<UserEvents> = null;
+    if (!this._current) {
+      const now = new Date();
+      // INITIAL STATE for UserEvents after first login by user
+      this._current = {
+        domainsSelected: [],
+        domainsAdded: [],
+        domainsRemoved: [],
+
+        strategiesAdded: [],
+        strategiesSelected: [],
+        strategiesRemoved: [],
+
+        resourcesFavorited: [],
+        resourcesRemovedFromFavorites: [],
+        resourcesHidden: [],
+        resourcesRemovedFromHidden: [],
+      };
```

Comment:
As discussed, please update this to an array of objects, with each object representing a single event. In Firebase, this should appear as a collection of events for each user.